src/rdpc/analysesProcessor.ts

```diff
@@ -277,3 +344,43 @@ const initializeRdpcInfo = (
     processingStatus: "REGISTERED",
   };
 };
+
+export const mergeDonorStateMaps = (
+  map: DonorRunStateMap,
+  mergeWith: DonorRunStateMap
+): DonorRunStateMap => {
+  const result = Object.entries(mergeWith).reduce<DonorRunStateMap>(
+    (acc, [donorId, rdpcInfo]) => {
+      acc[donorId] = {
+        publishedNormalAnalysis:
+          (acc[donorId]?.publishedNormalAnalysis || 0) +
+          rdpcInfo.publishedNormalAnalysis,
+        publishedTumourAnalysis:
+          (acc[donorId]?.publishedTumourAnalysis || 0) +
+          rdpcInfo.publishedTumourAnalysis,
+        alignmentsCompleted:
+          (acc[donorId]?.alignmentsCompleted || 0) +
+          rdpcInfo.alignmentsCompleted,
+        alignmentsRunning:
+          (acc[donorId]?.alignmentsRunning || 0) + rdpcInfo.alignmentsRunning,
+        alignmentsFailed:
+          (acc[donorId]?.alignmentsFailed || 0) + rdpcInfo.alignmentsFailed,
+        sangerVcsCompleted:
+          (acc[donorId]?.sangerVcsCompleted || 0) + rdpcInfo.sangerVcsCompleted,
+        sangerVcsRunning:
+          (acc[donorId]?.sangerVcsRunning || 0) + rdpcInfo.sangerVcsRunning,
+        sangerVcsFailed:
+          (acc[donorId]?.sangerVcsFailed || 0) + rdpcInfo.sangerVcsFailed,
+        totalFilesCount:
+          (acc[donorId]?.totalFilesCount || 0) + rdpcInfo.totalFilesCount,
+        filesToQcCount:
+          (acc[donorId]?.filesToQcCount || 0) + rdpcInfo.filesToQcCount,
+        releaseStatus: "NO_RELEASE",
+        processingStatus: "REGISTERED",
+      };
+      return acc;
+    },
+    { ...map }
+  );
+  return result;
+};
```

Comment:
If we changed the return types and implementation of the two counting functions to only include the subset of fields that they are responsible for, then this function can be as simple as a lodash merge, plus the two state fields.

We can remodel all the rdpc fields into two different types and define the combination as the union of the two, much like what we did with the clinical and rdpc split